databricks/koalas/plot.py

```diff
@@ -1118,6 +1119,173 @@ def _plot(data, x=None, y=None, subplots=False, ax=None, kind="line", **kwds):
+    """
+    from databricks.koalas import DataFrame, Series
+
+    data_preprocessor_map = {
+        "pie": TopNPlot().get_top_n,
+        "bar": TopNPlot().get_top_n,
+        "barh": TopNPlot().get_top_n,
+        "scatter": TopNPlot().get_top_n,
+        "area": SampledPlot().get_sampled,
+        "line": SampledPlot().get_sampled,
+    }
+    # make the arguments values of matplotlib compatible with that of plotting backend
+    args_map = {
+        "plotly": [
+            ("ax", None),
+            ("logx", "log_x"),
+            ("logy", "log_y"),
+            ("xlim", "range_x"),
+            ("ylim", "range_y"),
+            ("yerr", "error_y"),
+            ("xerr", "error_x"),
+        ]
+    }
+
+    if isinstance(data, Series):
+        positional_args = {
+            ("ax", None),
+            ("figsize", None),
+            ("use_index", True),
+            ("title", None),
+            ("grid", None),
+            ("legend", False),
+            ("style", None),
+            ("logx", False),
+            ("logy", False),
+            ("loglog", False),
+            ("xticks", None),
+            ("yticks", None),
+            ("xlim", None),
+            ("ylim", None),
+            ("rot", None),
+            ("fontsize", None),
+            ("colormap", None),
+            ("table", False),
+            ("yerr", None),
+            ("xerr", None),
+            ("label", None),
+            ("secondary_y", False),
+        }
+    elif isinstance(data, DataFrame):
+        positional_args = {
+            ("ax", None),
+            ("subplots", None),
+            ("sharex", None),
+            ("sharey", False),
+            ("layout", None),
+            ("figsize", None),
+            ("use_index", True),
+            ("title", None),
+            ("grid", None),
+            ("legend", True),
+            ("style", None),
+            ("logx", False),
+            ("logy", False),
+            ("loglog", False),
+            ("xticks", None),
+            ("yticks", None),
+            ("xlim", None),
+            ("ylim", None),
+            ("rot", None),
+            ("fontsize", None),
+            ("colormap", None),
+            ("table", False),
+            ("yerr", None),
+            ("xerr", None),
+            ("secondary_y", False),
+            ("sort_columns", False),
+        }
+    # removing keys that are not required
+    kwargs.pop("self", None)
+    kwargs.pop("kwds", None)
+    kwargs.pop("kind", None)
+
+    if backend_name in args_map:
+        for arg_name_mpl, arg_name_ply in args_map[backend_name]:
+            if arg_name_ply is None:
+                kwargs.pop(arg_name_mpl, None)
+            else:
+                kwargs[arg_name_ply] = kwargs.pop(arg_name_mpl)
+
+    # Values not being same as default implies user is explicitly passing the arguments
+    for arg, def_val in positional_args:
+        if arg in kwargs and kwargs[arg] == def_val:
+            kwargs.pop(arg, None)
```

Comment:
I guess we should apply this before Line 1274, otherwise ploty specific arguments could have different values?

Reply:
Made the changes. Now we try to map if possible and remove arguements if default.